### DISCUSSION TITLE
Don't confirm unstaging everything if there's nothing else unstaged already

### DISCUSSION THREAD

**Discussion by verhovsky** [magit/magit#5128] [Support] [answered]
If I stage all the files with changes in my repo and then press `u` on the "Staged changes (11)" line to unstage all of them again, magit will ask to confirm "unstage all changes? (y or n)". I don't see how this confirmation can be useful when I have nothing unstaged.

I can understand when there's a few staged and and a few unstaged files because unstaging everything destroys information/could undo all the work I did carefully staging pieces of files or whatever, but when everything is staged, unstaging everything shouldn't open a confirmation dialog because it's not a destructive operation.

**Comment by tarsius** [comment]
> I can understand when there's a few staged and and a few unstaged files because unstaging everything destroys information/could undo all the work I did carefully staging pieces of files or whatever, but when everything is staged, unstaging everything shouldn't open a confirmation dialog because it's not a destructive operation.

And that's how things are already supposed to work. Are you maybe using an old Magit version?

**Reply by verhovsky** [reply]
`b5637d66` I have untracked files, maybe that's what's causing it?

**Reply by tarsius** [reply]
Indeed,

```elisp
(defun magit-unstage-all ()
  "Remove all changes from the staging area."
  (interactive)
  (unless (magit-anything-staged-p)
    (user-error "Nothing to unstage"))
  (when (or (magit-anything-unstaged-p)
            (magit-untracked-files))        ; <===== here
    (magit-confirm 'unstage-all-changes))
  (magit-wip-commit-before-change nil " before unstage")
  (magit-run-git "reset" "HEAD" "--" magit-buffer-diff-files)
  (magit-wip-commit-after-apply nil " after unstage"))
```

**Reply by verhovsky** [reply]
but why? they're separate groups of changes and it won't affect it. in a lot of my repos i end up with untracked files perpetually, so this is effectively a `(when true)` for a lot of the time for me.

**Reply by tarsius** [reply, marked as answer]
If you have untracked files, then unstaging everything is potentially destructive. Some files might become untracked again.

**Reply by verhovsky** [reply]
makes sense. you could check if that's the case

**Reply by tarsius** [reply]
By the way, you can disable all or certain confirmations using option `magit-no-confirm`. Beside `stage-all-changes` and `unstage-all-changes`, you might also find `safe-with-wip` useful.
